.github/workflows/workflow-common.yml

```diff
@@ -61,18 +61,61 @@ jobs:
           docker system prune -af --volumes
           ./run.py -u
 
+      - name: Archive unit-tests results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: unit-test-report
+          path: build/test_doc
+  integration_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - esp32
+          - x86
+          - stm32
+          # optional target, should be added for new tests in case of introducing markers
+          # - not esp32 and not x86 and not stm32
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Apt install dependencies
+        run: |
+          sudo apt-get update --fix-missing
+          sudo apt-get install doxygen
+          sudo apt-get clean
+          sudo apt-get autoremove --yes
+          sudo rm -rf /var/lib/{cache,log}
+          sudo rm -rf /var/lib/apt/lists/*
+
+      - name: Pip install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install -r ci/requirements.txt
+
+      - name: Build docker image for tests purpose
+        # Clean docker before test (github like to go out of memory)
+        # Also there is need for repeatable results with docker (cache is faster but it can give some false positives and false negatives while testing)
+        run: |
+          docker system prune -af --volumes
+          ./run.py -p "new proj_${{ matrix.target }} --target ${{ matrix.target }}"
+          rm -rf proj_${{ matrix.target }}
+
       - name: Integration tests
+        # -k could be replaced with -m as markers will be introduced in integration tests in the future
```

Comment:
This comment should refering to pytest command options in run.py -> main -> if args.target (line 329)

Reply:
Right, my copypaste algorithm misses this comment when moving implementation in different place